### PR TITLE
authentication front end

### PR DESCRIPTION
**Pull request by jhubbard2017**
Made changes requested from previous pull request.

### CODE REVIEW

File: src/app/routes/auth/components/DisplayContent.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+
```
Comment (lmartin73) [inline src/app/routes/auth/components/DisplayContent.js:2]: Review verbiage on this page.

File: src/app/routes/auth/components/Footer.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+
+export default class Footer extends React.Component {
+
+    constructor() {
+        super();
+        // Initialize variables for action methods to keep component as a controlled component
+        this.facebook = this.facebookAction.bind(this);
```
Comment (lmartin73) [inline src/app/routes/auth/components/Footer.js]: use verbs when naming methods.

File: src/app/routes/auth/containers/Forgot.js [hunk 3]
```diff
@@ -1,125 +1,100 @@
...
+        // Initialize variable for forgot password action method to keep form as a controlled component
+        this.submitForm = this.forgotPasswordAction.bind(this);
+        this.displayForm = this.passwordRecoveryForm();
+    }
+
+    forgotPasswordAction(event) {
+        event.preventDefault();
+
+        const formValid = this.refs.email.value.length > 0 && this.refs.email.validity.valid;
+        if (formValid) {
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Forgot.js]: We should find a better way to validate forms.

File: src/app/routes/auth/containers/Forgot.js [hunk 4]
```diff
@@ -1,125 +1,100 @@
...
-                <div className="well no-padding">
-                  <form action="#/login" id="login-form" className="smart-form client-form">
-                    <header>
-                      Forgot Password
-                    </header>
 
+    constructor() {
+        super();
+        // Initialize variable for forgot password action method to keep form as a controlled component
+        this.submitForm = this.forgotPasswordAction.bind(this);
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Forgot.js]: Rename these to the similar method names

File: src/app/routes/auth/containers/LockedScreen.js [hunk 5]
```diff
@@ -1,48 +1,84 @@
...
-                  </div>
-                </div>
-                <p className="no-margin margin-top-5">
-                  Logged as someone else? <a href="#/login"> Click here</a>
-                </p>
-              </div>
+    constructor() {
+        super();
+        // Initialize variables for submit password action method to keep form as a controlled component
+        this.submitForm = this.submitPassword.bind(this);
```
Comment (lmartin73) [inline src/app/routes/auth/containers/LockedScreen.js]: names should be similar.

File: src/app/routes/auth/containers/LockedScreen.js [hunk 6]
```diff
@@ -1,48 +1,84 @@
...
+        this.getPhotoSrc = this.getPhotoSrc();
+    }
 
-            </div>
-            <p className="font-xs margin-top-5">
-              Copyright SmartAdmin 2014-2020.
+    submitPassword(event) {
+        event.preventDefault();
+
+        const formValid = this.refs.password.value.length >= 8;
```
Comment (lmartin73) [inline src/app/routes/auth/containers/LockedScreen.js]: we should find a way to validate forms different from this.

File: src/app/routes/auth/containers/LockedScreen.js [hunk 7]
```diff
@@ -1,48 +1,84 @@
...
+        return (
+            <div id="extr-page">
+                <div id="main" role="main">
+                    <UiValidate options={validationOptions}>
+                        <form className="lockscreen animated flipInY" onSubmit={this.submitForm}>
+                            <div className="logo">
+                                <div><h1 className="logo-name text-center">TrackIt+</h1></div>
+                            </div>
+                            <div>
+                                <img src={this.getPhotoSrc} alt="" width="120" height="120"/>
```
Comment (lmartin73) [inline src/app/routes/auth/containers/LockedScreen.js]: We should be storing data inside of states.

File: src/app/routes/auth/containers/LockedScreen.js [hunk 8]
```diff
@@ -1,48 +1,84 @@
...
+        const formValid = this.refs.password.value.length >= 8;
+        if (formValid) {
+            // Todo: Check user password here
+            // password value -> this.password.value
+        }
+    }
 
-            </p>
-          </form>
+    getPhotoSrc() {
```
Comment (lmartin73) [inline src/app/routes/auth/containers/LockedScreen.js]: Don't need this logic. will always be a photo url

File: src/app/routes/auth/containers/LockedScreen.js [hunk 9]
```diff
@@ -1,48 +1,84 @@
...
+                                        <small><i className="fa fa-lock text-muted"/> &nbsp;Locked</small>
+                                    </h1>
+                                    <p className="text-muted">Email here</p>
+                                    <div className="input-group">
+                                        <input className="form-control" type="password" name="password" ref="password" />
+                                        <div className="input-group-btn">
+                                            <button className="btn btn-primary" type="submit"><i className="fa fa-key"/></button>
+                                        </div>
+                                    </div>
+                                    <p className="no-margin margin-top-5">Logged as someone else? <a href="#/login"> Click here</a></p>
```
Comment (lmartin73) [inline src/app/routes/auth/containers/LockedScreen.js]: "Login as someone else?"

File: src/app/routes/auth/containers/Login.js [hunk 10]
```diff
@@ -1,122 +1,120 @@
...
+export default class Login extends React.Component {
 
-          <div id="content" className="container">
-            <div className="row">
-              <div className="col-xs-12 col-sm-12 col-md-7 col-lg-8 hidden-xs hidden-sm">
-                <h1 className="txt-color-red login-header-big">SmartAdmin</h1>
+    constructor() {
+        super();
+        // Initialize variable for sign in action method to keep form as a controlled component
+        this.submitForm = this.signInAction.bind(this);
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Login.js]: make method alias similar names.

File: src/app/routes/auth/containers/Login.js [hunk 11]
```diff
@@ -1,122 +1,120 @@
...
-                      <a href="#/smartadmin/different-versions.html" className="btn btn-danger btn-sm">Find out more</a>
-                    </div>
-                  </div>
-                  <img src="assets/img/demo/iphoneview.png" className="pull-right display-image" alt=""
-                       style={{width: '210px'}}/>
-                </div>
-                <div className="row">
-                  <div className="col-xs-12 col-sm-12 col-md-6 col-lg-6">
-                    <h5 className="about-heading">About SmartAdmin - Are you up to date?</h5>
+        const formValid = this.refs.email.value.length != 0 && this.refs.email.validity.valid &&
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Login.js]: find a way to do form validation logic.

File: src/app/routes/auth/containers/Register.js [hunk 12]
```diff
@@ -1,211 +1,183 @@
...
-                          <b className="tooltip tooltip-bottom-right">Don't forget your password</b> </label>
-                      </section>
-                    </fieldset>
+export default class Register extends React.Component {
 
+    constructor() {
+        super();
+        // Initialize variable for register action method to keep form as a controlled component
+        this.submitForm = this.createAccountAction.bind(this);
+        this.displayForm = this.registrationForm();
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Register.js]: make method alias similar

File: src/app/routes/auth/containers/Register.js [hunk 13]
```diff
@@ -1,211 +1,183 @@
...
+        this.displayTerms = this.termsModalView();
+    }
+
+    createAccountAction(event) {
+        event.preventDefault();
+
+        const formValid = this.refs.firstname.value.length > 0 && this.refs.lastname.value.length > 0 &&
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Register.js]: find the validation api to use

**Review by lmartin73 — CHANGES_REQUESTED** [review]

File: src/app/routes/auth/containers/Forgot.js [hunk 14]
```diff
@@ -1,125 +1,93 @@
...
-                <div className="well no-padding">
-                  <form action="#/login" id="login-form" className="smart-form client-form">
-                    <header>
-                      Forgot Password
-                    </header>
 
+    constructor() {
+        super();
+        // Initialize variable for forgot password action method to keep form as a controlled component
+        this.displayForm = this.displayForm();
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Forgot.js]: bind the this pointer to this method.

File: src/app/routes/auth/containers/Login.js [hunk 15]
```diff
@@ -1,122 +1,111 @@
...
-                  </div>
-                  <img src="assets/img/demo/iphoneview.png" className="pull-right display-image" alt=""
-                       style={{width: '210px'}}/>
-                </div>
-                <div className="row">
-                  <div className="col-xs-12 col-sm-12 col-md-6 col-lg-6">
-                    <h5 className="about-heading">About SmartAdmin - Are you up to date?</h5>
+    constructor(props) {
+        super(props);
+        this.displayForm = this.displayForm();
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Login.js]: bind the this pointer to this method.

File: src/app/routes/auth/containers/Register.js [hunk 16]
```diff
@@ -1,211 +1,172 @@
...
-
-                  <form action="#/dashboard" id="smart-form-register" className="smart-form client-form">
-                    <header>
-                      Registration is FREE*
-                    </header>
 
+    constructor() {
+        super();
+        // Initialize variable for register action method to keep form as a controlled component
+        this.displayForm = this.displayForm();
```
Comment (lmartin73) [inline src/app/routes/auth/containers/Register.js]: should you bind this this pointer to these methods?

**Review by lmartin73 — CHANGES_REQUESTED** [review]

File: src/app/routes/auth/containers/Login.js (hunk 15 above)
Comment (jhubbard2017) [reply src/app/routes/auth/containers/Login.js]: Done

File: src/app/routes/auth/containers/Register.js (hunk 16 above)
Comment (jhubbard2017) [reply src/app/routes/auth/containers/Register.js]: Done

**Review by lmartin73 — APPROVED** [review]

**Pull request merged**